sos/report/plugins/charmed_mysql_k8s.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.#
+
+import os
+import tempfile
+
+from sos.report.plugins import Plugin, PluginOpt, UbuntuPlugin, DebianPlugin
+
+
+class CharmedMySQLK8s(Plugin, UbuntuPlugin, DebianPlugin):
+    """
+    The Charmed MySQL K8s plugin is used to collect MySQL configuration and
+    logs from the Charmed MySQL on Kubernets. It also collects MySQL Router
+    and MySQL Shell configuration and logs where available,
+    journal logs for the snap, and snap info.
+
+    If the `dumpdbs` option is set to `True`, the plugin will also try and
+    collect the names of the databases that the user has access to. The
+    `mysql` user is used by default, but that can be set using the `dbuser`
+    option. When using the `dumpdbs` option, you must then provide the
+    password for the user using the `dbpass` option.
```

Comment:
Minor docstring nit: the plugin also supports obtaining the password from `MYSQL_PWD` if set, but it's not a big deal.